test/gtest/ucs/test_event_set.cc

```diff
@@ -0,0 +1,171 @@
+/**
+* Copyright (C) Hiroyuki Sato. 2019. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#include <common/test.h>
+extern "C" {
+#include <ucs/sys/event_set.h>
+#include <pthread.h>
+}
+
+#define MAX_BUF_LEN 255
+
+static const char *UCS_EVENT_SET_TEST_STRING = "ucs_event_set test string";
+
+class test_event_set : public ucs::test {
+public:
+    static const char *evfd_data;
+
+protected:
+    static void* event_set_read_func(void *arg) {
+        int *fd = (int *)arg;
+        usleep(10000);
+        write(fd[1], evfd_data, strlen(test_event_set::evfd_data));
+        return 0;
+    }
+
+    static void* event_set_tmo_func(void *arg) {
+        usleep(50);
+        return 0;
+    }
+};
+
+const char *test_event_set::evfd_data = UCS_EVENT_SET_TEST_STRING;
+
+static ucs_status_t event_set_func1(int fd,
+                                    int events)
+{
+    char buf[MAX_BUF_LEN];
+    int n;
+    memset(buf, 0, MAX_BUF_LEN);
+
+    EXPECT_EQ(UCS_EVENT_SET_EVREAD, events);
+    n = read(fd, buf, MAX_BUF_LEN);
+    if (n==-1) {
+        return UCS_ERR_IO_ERROR;
+    }
+    EXPECT_TRUE(strcmp(UCS_EVENT_SET_TEST_STRING, buf) == 0);
+    return UCS_OK;
+}
+
+static ucs_status_t event_set_func2(int fd, int events)
+{
+    EXPECT_EQ(UCS_EVENT_SET_EVWRITE, events);
+    return UCS_OK;
+}
+
+// timeout doesn't call event handler
+// raise an Error if this function called.
+static ucs_status_t event_set_func3(int fd, int events)
+{
+//    TODO FAIL() desn't work in this function.
+//    FAIL();
+    return UCS_ERR_IO_ERROR;
```

Comment:
Could you tell me how to use `FAIL()` functions? timeout doesn't callback function. That's why this function shouldn't call. If this callback called, It should fail.

Reply:
i think `ADD_FAILURE()` should work